frontend/src/components/common/button/button.tsx

```diff
@@ -1,36 +1,51 @@
 import { AppRoute } from 'common/enums/enums';
-import { FC } from 'common/types/types';
+import { FC, IconName } from 'common/types/types';
 import { getValidClasses } from 'helpers/helpers';
+import { ReactElement } from 'react';
 
-import { Link } from '../common';
+import { Icon, Link } from '../common';
 import styles from './styles.module.scss';
 
 type Props = {
-  label: string;
-  btnColor?: 'blue' | 'gray';
+  label?: string;
+  btnColor?: 'blue' | 'gray' | 'transparent';
   type?: 'button' | 'submit';
-  btnType?: 'filled' | 'outlined';
-  to?: AppRoute;
+  btnType?: 'filled' | 'outlined' | 'icon';
+  to?: AppRoute | string;
   onClick?: () => void;
+  iconName?: IconName;
 };
 
 const Button: FC<Props> = ({
   btnType = 'filled',
   type = 'button',
-  btnColor = 'blue',
+  btnColor = 'transparent',
   label,
   to,
   onClick,
+  iconName,
 }) => {
   const isLink = Boolean(to);
 
+  const getContent = (): ReactElement | string => {
+    const hasIcon = Boolean(iconName);
+
+    if (hasIcon) {
+      return (<Icon name={iconName as IconName} />) as ReactElement;
+    }
+
+    return label as string;
+  };
+
   if (isLink) {
     return (
       <Link
         to={to as AppRoute}
-        className={getValidClasses(styles.button, styles[`button-${btnType}`])}
+        className={getValidClasses(
+          btnType !== 'icon' && [styles.button, styles[`button-${btnType}`]],
```

Comment:
```suggestion
         [styles.button, styles[`button-${btnType}`]],
```
by default, the button should be without any colors. apply them **only** by `btnStyle` and `btnColor` props

Reply:
isn't fixed. we don't need this condition